apps/hash-api/src/graphql/resolvers/integrations/linear/sync-workspaces-with-teams.ts

```diff
@@ -18,24 +20,55 @@ export const syncLinearIntegrationWithWorkspacesMutation: ResolverFn<
 > = async (
   _,
   { linearIntegrationEntityId, syncWithWorkspaces },
-  { dataSources, user },
+  { dataSources, user, temporal, vault },
 ) => {
+  if (!vault) {
+    throw new Error("Vault client not available");
+  }
+
+  if (!temporal) {
+    throw new Error("Temporal client not available");
+  }
+
   const linearIntegration = await getLinearIntegrationById(dataSources, {
     entityId: linearIntegrationEntityId,
   });
 
+  const userAccountId = extractOwnedByIdFromEntityId(
+    linearIntegration.entity.metadata.recordId.entityId,
+  );
+
+  const linearUserSecret = await getLinearUserSecretByLinearOrgId(dataSources, {
+    userAccountId,
+    linearOrgId: linearIntegration.linearOrgId,
+  });
+
+  const vaultSecret = await vault.read<{ value: string }>({
+    secretMountPath: "secret",
+    path: linearUserSecret.vaultPath,
+  });
+
+  const apiKey = vaultSecret.data.value;
+  const linearClient = new Linear({
+    temporalClient: temporal,
+    apiKey,
+  });
+
   await Promise.all(
-    syncWithWorkspaces.map(async ({ workspaceEntityId, linearTeamIds }) =>
-      syncLinearIntegrationWithWorkspace(dataSources, {
+    syncWithWorkspaces.map(async ({ workspaceEntityId, linearTeamIds }) => {
+      await linkIntegrationToWorkspace(dataSources, {
         linearIntegrationEntityId,
         workspaceEntityId,
         linearTeamIds,
         actorId: user.accountId,
-      }),
-    ),
+      });
+      return linearClient.syncWorkspace({
```

Comment:
I think we shouldn't return this, because it could take a long time and the frontend waits for this return.

We'll need some way of telling the user the status of their sync, which we can add later
```suggestion
      linearClient.syncWorkspace({
```